### PR TITLE
#1007 Do not bind session as a side effect when removing pages

### LINKED ISSUE #1007: SessionQuotaManagingDataStore should not bind session when removing pages
`SessionQuotaManagingDataStore` currently binds the session as a side-effect when `removeAllPages` or `removePage` is called.

Both methods call `IPageContext.getSessionData(key, supplier)` which in the case of `DefaultPageContext`, binds the session:

https://github.com/apache/wicket/blob/a20a38c0068b4f926386bfb2c9dd46a470ff4baa/wicket-core/src/main/java/org/apache/wicket/pageStore/DefaultPageContext.java#L77-L89

The `removePage` and `removeAllPages` methods should probably do nothing if the session has not been bound yet.

### PR DESCRIPTION
**Pull request by theigl**
This PR passes `null` instead of a  null-supplier when pages are removed from a `SessionQuotaManagingDataStore`. This prevents binding the session as a side-effect.

Resolves #1007.

### CODE REVIEW

File: datastores-parent/datastore-tests/pom.xml [hunk 1]
```diff
@@ -76,5 +76,10 @@
 			<artifactId>websocket-jakarta-server</artifactId>
 			<scope>compile</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<scope>compile</scope>
```
Comment (solomax) [inline datastores-parent/datastore-tests/pom.xml:82]: please drop this `compile` scope (default `test` will be used in this case :)

Comment (martin-g) [inline datastores-parent/datastore-tests/pom.xml:82]:
```suggestion
```

**Review by martin-g — APPROVED** [review]

File: datastores-parent/datastore-tests/pom.xml [hunk 2]
```diff
@@ -76,5 +76,9 @@
 			<artifactId>websocket-jakarta-server</artifactId>
 			<scope>compile</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
```
Comment (martin-g) [inline datastores-parent/datastore-tests/pom.xml:81]:
```suggestion
			<artifactId>mockito-core</artifactId>
			<scope>test</scope>
```

File: datastores-parent/datastore-tests/pom.xml (hunk 1 above)
Comment (martin-g) [reply datastores-parent/datastore-tests/pom.xml]: The tests are in `datastores-parent/datastore-tests/src/main/java/...` I do not remember why it has been done this way. 
The build fails now.
IMO the tests should be moved to `src/test/java/` but reverting the scope to `compile` is OKish too for this PR!

Comment (theigl) [reply datastores-parent/datastore-tests/pom.xml]: Exactly. It does not work with `test` scope at the moment.

**Comment by solomax** [comment]
My bad :(
I didn't noticed this is separate `tests` module

the scope should be `compile`

sorry for the noise :(

File: datastores-parent/datastore-tests/pom.xml (hunk 1 above)
Comment (solomax) [reply datastores-parent/datastore-tests/pom.xml]: My bad :(
I didn't noticed this is separate `tests` module

the scope should be `compile`

sorry for the noise :(

**Review by solomax — APPROVED** [review]

**Comment by theigl** [comment]
@martin-g: Can I revert your changes to the POM so the build passes again?

**Comment by martin-g** [comment]
Sure!

**Comment by martin-g** [comment]
@solomax Nice commit message "Ыуеештп ьщслшещ ысщзу ещ иу сщьзшду " :-)

**Comment by solomax** [comment]
> @solomax Nice commit message "Ыуеештп ьщслшещ ысщзу ещ иу сщьзшду " :-)

:)))
Sorry, forgot to switch keyboard back to English :)))

**Pull request merged**